src/views/Player.vue

```diff
@@ -376,13 +396,28 @@ export default {
 
     async setPlayerProperties(player) {
       player.on("ready", () => {
+        // start playing from 5 seconds before where the user left off in previous session
+        if (this.previousPlayerTime > stepBackTime){
+          this.player.currentTime = this.previousPlayerTime - stepBackTime
+        }
+
+        // start from beginning if video was watched till the end in the last session
+        if (this.player.duration - this.previousPlayerTime <= 2) {
+          this.player.currentTime = 0
+        }
```

Comment:
Eariler the code was
```if(this.player.previousPlayerTime == this.player.duration)
        this.player.currentTime = 0```
But while testing I realised that the `currentTime` or `previousPlayerTime` will never be exactly equal to the `player.duration`. So kept 2 seconds as buffer

Reply:
Make `2` a constant. `START_BUFFER_TIME`